src/rust/src/note_encryption.rs

```diff
@@ -0,0 +1,94 @@
+use group::GroupEncoding;
+use zcash_note_encryption::{Domain, EphemeralKeyBytes, ShieldedOutput, ENC_CIPHERTEXT_SIZE};
+use zcash_primitives::{
+    consensus::BlockHeight,
+    memo::MemoBytes,
+    sapling::{
+        self,
+        note_encryption::{PreparedIncomingViewingKey, SaplingDomain},
+        SaplingIvk,
+    },
+};
+
+use crate::{params::Network, wallet_scanner::ffi::SaplingShieldedOutput};
+
+pub(crate) fn try_sapling_note_decryption(
```

Comment:
Consider renaming this to avoid confusion with `sapling::note_encryption::try_sapling_note_decryption`.

Reply:
The two cannot be confused in Rust because their argument types are different. But put another way, these methods _are_ the same method; the only reason this one exists is to convert between FFI-compatible types and the Rust API. So we want the C++ method name to be the same as the Rust method name.